pandas-stubs/io/sql.pyi

```diff
@@ -1,191 +1,139 @@
-    columns: Sequence[str] = ...,
-    chunksize: int = ...,
+    params: list[str] | dict[str, str] | None = ...,
+    parse_dates: list[str] | dict[str, str] | dict[str, dict[str, Any]] | None = ...,
+    columns: list[str] = ...,
+    *,
+    chunksize: int,
+) -> Iterator[DataFrame]: ...
+@overload
+def read_sql(
+    sql: str,
+    con: Any,
+    index_col: str | list[str] | None = ...,
+    coerce_float: bool = ...,
+    params: list[str] | dict[str, str] | None = ...,
+    parse_dates: list[str] | dict[str, str] | dict[str, dict[str, Any]] | None = ...,
+    columns: list[str] = ...,
+    chunksize: None = ...,
 ) -> DataFrame: ...
-def to_sql(
-    frame,
-    name,
-    con,
-    schema=...,
-    if_exists: str = ...,
-    index: bool = ...,
-    index_label=...,
-    chunksize=...,
-    dtype=...,
-    method=...,
-) -> None: ...
-def has_table(table_name, con, schema=...): ...
-
-table_exists = has_table
-
-def pandasSQL_builder(con, schema=..., meta=..., is_cursor: bool = ...): ...
-
-class SQLTable(PandasObject):
-    name = ...
-    pd_sql = ...
-    prefix = ...
-    frame = ...
-    index = ...
-    schema = ...
-    if_exists = ...
-    keys = ...
-    dtype = ...
-    table = ...
-    def __init__(
-        self,
-        name,
-        pandas_sql_engine,
-        frame=...,
-        index: bool = ...,
-        if_exists: str = ...,
-        prefix: str = ...,
-        index_label=...,
-        schema=...,
-        keys=...,
-        dtype=...,
-    ) -> None: ...
-    def exists(self): ...
-    def sql_schema(self): ...
-    def create(self) -> None: ...
-    def insert_data(self): ...
-    def insert(self, chunksize=..., method=...) -> None: ...
-    def read(
-        self, coerce_float: bool = ..., parse_dates=..., columns=..., chunksize=...
-    ): ...
 
 class PandasSQL(PandasObject):
-    def read_sql(self, *args, **kwargs) -> None: ...
+    def read_sql(self, *args, **kwargs): ...
     def to_sql(
         self,
-        frame,
-        name,
+        frame: DataFrame,
+        name: str,
         if_exists: str = ...,
         index: bool = ...,
         index_label=...,
         schema=...,
         chunksize=...,
-        dtype=...,
+        dtype: DtypeArg | None = ...,
         method=...,
-    ) -> None: ...
+    ) -> int | None: ...
 
-class SQLDatabase(PandasSQL):
-    connectable = ...
-    meta = ...
-    def __init__(self, engine, schema=..., meta=...) -> None: ...
-    def run_transaction(self) -> None: ...
-    def execute(self, *args, **kwargs): ...
-    def read_table(
-        self,
-        table_name,
-        index_col=...,
-        coerce_float: bool = ...,
-        parse_dates=...,
-        columns=...,
-        schema=...,
-        chunksize=...,
-    ): ...
-    def read_query(
-        self,
-        sql,
-        index_col=...,
-        coerce_float: bool = ...,
-        parse_dates=...,
-        params=...,
-        chunksize=...,
-    ): ...
-    def to_sql(
+class SQLTable(PandasObject):
```

Comment:
not in docs, so delete

Reply:
`SQLTable` is needed to provide typing for the `Callable` version of `to_sql`. Moreover, attributes of `SQLTable` are needed in this method. There is an example that uses `SQLTable` and some attributes.